gensim/summarization/bm25.py

```diff
@@ -80,33 +76,47 @@ def __init__(self, corpus):
         """
         self.corpus_size = len(corpus)
         self.avgdl = sum(float(len(x)) for x in corpus) / self.corpus_size
-        self.corpus = corpus
-        self.f = []
-        self.df = {}
+        self.doc_freqs = []
         self.idf = {}
         self.doc_len = []
-        self.initialize()
+        self._initialize(corpus)
 
-    def initialize(self):
+    def _initialize(self, corpus):
         """Calculates frequencies of terms in documents and in corpus. Also computes inverse document frequencies."""
-        for document in self.corpus:
-            frequencies = {}
+
+        nd = {} # word -> number of documents with word
+        for document in corpus:
             self.doc_len.append(len(document))
+            frequencies = {}
             for word in document:
                 if word not in frequencies:
                     frequencies[word] = 0
                 frequencies[word] += 1
-            self.f.append(frequencies)
+            self.doc_freqs.append(frequencies)
 
             for word, freq in iteritems(frequencies):
-                if word not in self.df:
-                    self.df[word] = 0
-                self.df[word] += 1
-
-        for word, freq in iteritems(self.df):
-            self.idf[word] = math.log(self.corpus_size - freq + 0.5) - math.log(freq + 0.5)
-
-    def get_score(self, document, index, average_idf):
+                if word not in nd:
+                    nd[word] = 0
+                nd[word] += 1
+
+        # collect idf sum to calculate an average idf for epsilon value
+        idf_sum = 0
```

Comment:
Safer to use float (or explicitly cast to float below).

Both to avoid potential Python errors in integer vs float division in future refactorings, as well as (mainly) to make the intent clear to readers.

Reply:
oh, yes. you're right. Damn python2.